crypto/x509/x509_def.c

```diff
@@ -8,27 +8,125 @@
  */
 
 #include <stdio.h>
+#include "internal/e_os.h"
 #include "internal/cryptlib.h"
 #include <openssl/crypto.h>
 #include <openssl/x509.h>
 
+#if defined(_WIN32)
+static char *x509_private_dir = NULL;
+static char *x509_cert_area = NULL;
+static char *x509_cert_dir = NULL;
+static char *x509_cert_file = NULL;
+static char *openssldir = NULL;
+
+static char *get_windows_openssldir()
+{
+    DWORD keysize;
+    DWORD ktype;
+    HKEY hkey;
+    LSTATUS ret;
+    LPSTR tempstr = NULL;
+
+    if (openssldir != NULL)
+        return openssldir;
+
+    ret = RegOpenKeyEx(HKEY_LOCAL_MACHINE, TEXT("SOFTWARE\\Wow6432Node\\OpenSSL"), 0, KEY_QUERY_VALUE, &hkey);
```

Comment:
If the registry key has not been created, then we're never going to set `openssldir` - so it will continue to be NULL - and we're going to do this *every* time we need to use it. Don't we need to short circuit this if we've done it once - even if the result was "its not there"